x-pack/plugins/maps/public/routes/map_page/saved_map/saved_map.ts

```diff
@@ -340,6 +342,15 @@ export class SavedMap {
         },
       });
       return;
+    } else if (dashboardId) {
+      this._getStateTransfer().navigateToWithEmbeddablePackage('dashboards', {
```

Comment:
`navigateToWithEmbeddablePackage` is async. Might be good to throw an await in front of this call and the one on 337